### PR TITLE
Help about groups: Update of the whole text

### PR DESCRIPTION
**Pull request by mlep**
This is a major rewrite of the help page about groups.

I need help of users for reviewing the changes.
Some specific parts needs attention. They are indicated by "%%%Help needed".

Notes:
- Screenshots will be added, after merging, using gitbook.
- Text layout will be adapted using gitbook (after merging, so).

### CODE REVIEW

File: en/finding-sorting-and-cleaning-entries/groups.md [hunk 1]
```diff
@@ -1,132 +1,268 @@
...
+The **+** button (near the top of the pane) lets you create a new subgroup of the group **All Entries**, regardless of the currently selected group(s).
+The context menu also allows removing groups and/or subgroups, and to sort subgroups alphabetically.
+Moving groups to a different location in the tree can be done by drag-and-drop.
 
-Just like directories, groups are structured like a tree, with the group **All Entries** at the root. By right-clicking on a group you can add a new group to the tree, either at the same level as the selected group or as a subgroup of it. The **New Group** button lets you create a new subgroup of the group **All Entries**, regardless of the currently selected group\(s\). The context menu also allows to remove groups and/or subgroups, to sort subgroups alphabetically, or to move groups to a different location in the tree. The latter can also be done by Drag and Drop, with the restriction that Drag and Drop does not support changing the order of a group's subgroups.
+Undo and redo is supported for all edits. **%%%Help needed: Not true in 5.2. Should this be considered as a bug or a feature?%%%**
```
Comment (tobiasdiez) [inline en/finding-sorting-and-cleaning-entries/groups.md:58]: It's a known bug. Maybe remove the sentence about undo/redo completely since it's not that important?

File: en/finding-sorting-and-cleaning-entries/groups.md [hunk 2]
```diff
@@ -1,132 +1,268 @@
...
+For example, consider a supergroup containing entries with the keyword *distribution* and a subgroup containing entries with the keyword *gauss*.
+With the subgroup *gauss* defined as an intersection (of its supergroup), selecting the subgroup *gauss* displays entries that match both conditions, i.e. are concerned with Gaussian distributions.
+Note that entries that only belong to the subgroup *gauss* will not be shown, i.e. for an entry to be displayed when selecting *gauss*, it must be assigned to both the subgroup *gauss* and the supergroup *distribution*.
+By adding another intersection group for *laplace* to the supergroup *distribution*, the grouping can easily be extended to Laplace distributions.
+
+### Union between a group and its subgroups
+
+The union of a group and its subgroups is the logical complement of the intersection: when defined as *union*, selecting the group displays *both* the group's own entries and its subgroups' entries.
+
+**%%%Help needed: Add an example of a sensible use here. %%%**
```
Comment (tobiasdiez) [inline en/finding-sorting-and-cleaning-entries/groups.md:111]: Parent group: some topic (say music)
Subgroups: some subtopic (say classical muscic, Jazz, ...)

Than you can simply add articles dealing with a specific aspect to the corresponding subtopic, and they automatically appear in the topic group as well (without additional work).

File: en/finding-sorting-and-cleaning-entries/groups.md [hunk 3]
```diff
@@ -1,132 +1,268 @@
...
+**%%% screenshot here: grab of the group type %%%**
 
-#### Dynamic groups
+After creating an explicit-selection group, you select the entries to be assigned to it, and use either drag-and-drop or the context menu **Add selected entries to this group** of the group interface.
 
-The content of a dynamic group is defined by a logical condition. Only entries that meet this condition are contained in the group. This method uses the information stored in the database itself, and updates dynamically whenever the database changes.
+To remove entries from an explicit-selection group, select them and use the context menu **Remove selected entries from this group** of the group interface.
 
-Two types of conditions can be used:
+**%%%Help needed: Relevant? True for this entry type only? %%%**
```
Comment (tobiasdiez) [inline en/finding-sorting-and-cleaning-entries/groups.md:150]: Yes, you need to have uniquce citation keys. It's somewhat relevant as people were confused about it in the past. I would add it as an info-box (in gitbook).

File: en/finding-sorting-and-cleaning-entries/groups.md [hunk 4]
```diff
@@ -1,132 +1,268 @@
...
+The content of the group is updated dynamically whenever the database changes.
+
+#### Using the cited entries of a LaTeX document
+
+The group contains the entries cited in a LaTeX document, based on its '.aux' file.
+The .aux file has to be specified.
+
+**%%% screenshot here: grab of the group type %%%**
+
+**%%%Help needed: True? When updated exactly? %%%**
```
Comment (tobiasdiez) [inline en/finding-sorting-and-cleaning-entries/groups.md:209]: True. It's updated as soon as the aux file changes on the disk, i.e. usually after you run your latex compilation.

File: en/finding-sorting-and-cleaning-entries/groups.md [hunk 5]
```diff
@@ -1,132 +1,268 @@
...
+This is similar to the above, but rather than search for a single search term on a single field, a [search expression syntax](search.md#advanced) can be used.
+It supports logical operators (`AND`, `OR`, `NOT`) and allows searching multiple fields.
 
-When multiple groups are selected, either the union or the intersection of their content is shown, depending on the current settings. This allows to quickly combine multiple conditions, e.g. if you have a static group `Extremely Important` to which you assign all extremely important entries, you can view the extremely important entries in any other group by selecting both groups \(this requires to have **Intersection** selected in the settings\).
+**%%% screenshot here: grab of the group type %%%**
+
+For example, the search expression `keywords=regression and not keywords=linear` groups entries concerned with non-linear regression.
+
+The content of the group is updated dynamically whenever the database changes.
```
Comment (tobiasdiez) [inline en/finding-sorting-and-cleaning-entries/groups.md:182]: This sentence applies to all groups.

File: en/finding-sorting-and-cleaning-entries/groups.md [hunk 6]
```diff
@@ -1,132 +1,268 @@
...
+
+#### Specified keywords
+
+**%%%Help needed: I do not understand this feature. Please, step in %%%**
+
+**%%%Help needed: The following might be relevant %%%**
+By clicking the **Automatically create groups for database** button, you can quickly create a set of groups appropriate for your database.
```
Comment (tobiasdiez) [inline en/finding-sorting-and-cleaning-entries/groups.md:189]: It's no longer a buton, but anthor type of group. If you have an entry with `keywords = {A, B}`, then this group creates subgroups `A` and `B` both containing the entry.

File: en/finding-sorting-and-cleaning-entries/groups.md [hunk 7]
```diff
@@ -1,132 +1,268 @@
...
+The search can be case-sensitive or not (checkbox 'Case sensitive').
+The search can either be done as a plain-text or a regular-expression search (checkbox 'Regular expression').
 
-The logical complement to a refining group is a group that **includes its subgroups**, i.e. when selected, not only the group's own entries, but also its subgroups' entries are shown. In the groups tree, this type of group has a special icon \(this can be turned off in the preferences\).
+Obviously, this will work only for entries including the specified grouping field, and the quality of the grouping will depend on the content accuracy.
 
-In other words, this defines which entries are displayed when this group is selected:
+**%%%Help needed: True in 5.2? I do not understand the paragraph... So, if true in 5.2 and unclear to you please reword %%%**
+In the former case, JabRef allows to manually assign/remove entries to/from the group by simply appending/removing the search term to/from the content of the grouping field.
```
Comment (tobiasdiez) [inline en/finding-sorting-and-cleaning-entries/groups.md:168]: I think what this means is that JabRef will show the entry automatically in a group that matches entries with keyword `A` if you add `A` in the keyword field of that entry. This is not specific to this group: as soon as an entry is changed so that it now matches the criteria of a group, it's added to that group automatically.

**Review by tobiasdiez — COMMENTED** [review]
Nice, this rework was really necessary. The current information is almost from the stone age. 

I've added a few answers to your questions.

File: en/finding-sorting-and-cleaning-entries/groups.md (hunk 1 above)
Comment (mlep) [reply en/finding-sorting-and-cleaning-entries/groups.md:58]: Sentence removed.

File: en/finding-sorting-and-cleaning-entries/groups.md (hunk 2 above)
Comment (mlep) [reply en/finding-sorting-and-cleaning-entries/groups.md:111]: Thanks for the suggestion. Added!

File: en/finding-sorting-and-cleaning-entries/groups.md (hunk 3 above)
Comment (mlep) [reply en/finding-sorting-and-cleaning-entries/groups.md:150]: OK, I will add an infobox here, after merging, using gitbook.

File: en/finding-sorting-and-cleaning-entries/groups.md (hunk 4 above)
Comment (mlep) [reply en/finding-sorting-and-cleaning-entries/groups.md:209]: Well, I tested it and could not find out when it was updated (except after restarting JabRef). Unfortunately, a latex compilation is not enough...

File: en/finding-sorting-and-cleaning-entries/groups.md (hunk 5 above)
Comment (mlep) [reply en/finding-sorting-and-cleaning-entries/groups.md:182]: I prefer to keep the sentence for each cases because I guess users do not read the help from the beginning to the end. I suppose they are likely to go to the kind of group they are interested in, and so I choose to repeat the sentence so that they find the information directly.

File: en/finding-sorting-and-cleaning-entries/groups.md (hunk 6 above)
Comment (mlep) [reply en/finding-sorting-and-cleaning-entries/groups.md:189]: OK, thanks. Modified!

File: en/finding-sorting-and-cleaning-entries/groups.md (hunk 7 above)
Comment (mlep) [reply en/finding-sorting-and-cleaning-entries/groups.md:168]: OK, I kept a modified version of it.

**Comment by mlep** [comment]
Thank @tobiasdiez for your encouragements and comments.
I believe groups are a key feature of JabRef (at least, it is for me!), so I thought the update was really needed here.

File: en/finding-sorting-and-cleaning-entries/groups.md [hunk 8]
```diff
@@ -1,132 +1,259 @@
...
 
-## Advanced features
+When selecting multiple groups, you can choose to:
 
-After mastering the grouping concepts described above, the following advanced features might come in handy.
+- display only entries belonging to all selected groups (intersection)
+- display all entries belonging to one or more of the selected groups (union). This is the default option.
 
-### Automatically creating dynamic groups
+### Automatically assign new entry go selected grouping
```
Comment (koppor) [inline en/finding-sorting-and-cleaning-entries/groups.md:238]:
```suggestion
### Automatically assign new entry to selected grouping
```

**Comment by mlep** [comment]
@koppor  Thanks for reviewing!
